### ISSUE TITLE
runscript gives error at the end of a run at ECMWF.atos

### ISSUE THREAD

**Issue by tsemmler05** [esm-tools/esm_tools#1052]
When running a day simulation on ECMWF.atos, the simulation finishes normally but files are not moved into the correct directories because of an error message from the esm_tools:

usage: esm_runscripts [-h] [-d] [-v] [--contained-run] [--open-run] [-e EXPID]
                      [-c] [-P] [--modify-config MODIFY] [-j LAST_JOBTYPE]
                      [-t TASK] [-i INSPECT] [-p PID] [-s START_DATE]
                      [-x EXCLUDE] [-o ONLY] [-r RUN_NUMBER] [-U]
                      [--update-filetypes UPDATE_FILETYPES [UPDATE_FILETYPES ...]]
                      [--no-motd] [--ignore-config-warnings]
                      runscript
esm_runscripts: error: argument -p/--pid: expected one argument

This happens when I use the ecmwf-atos branch of the esm_tools version 6.22.0

**Comment by nwieters** [comment]
Hi @tsemmler05,

can you give the path to your experiment, please. I am looking at something related to this. 
The value for the argument -p/--pid is set/derived in the *.run file. This file can be found in the experiment folder in `<experimentid>/scripts`:
`process=$!`
Maybe it is something going wrong with this command and a wrong or empty `pid` is submitted to the following esm_runscipts call.

**Comment by tsemmler05** [comment]
Hi @nwieters!

This is the experiment folder on ECMWF.atos:

/scratch/duts/runtime/awicm3-frontiers-xios/OMP1/

It is the second of the two log files that threw the error message. Thanks for looking into this!

**Comment by tsemmler05** [comment]
Hi again!

I decided to put in a fresh run under /scratch/duts/runtime/awicm3-frontiers-xios/testpidbug/

Maybe this is easier to investigate as there is no mix with the other things I am testing at the moment. The end of the run script *.run looks like this:

time srun -l --kill-on-bad-exit=1 --cpu_bind=none --multi-prog hostfile_srun 2>&
1 &

# Call to esm_runscript to start subjobs:
# ['tidy']
process=$! 
# Comment the following line if you don't want esm_runscripts to restart:
cd /scratch/duts/runtime/awicm3-frontiers-xios//testpidbug/scripts/
esm_runscripts awicm3-frontiers-xios-ecmwf-atos-TCO319L137-DART_initial.yaml -e 
testpidbug -t observe_compute -p ${process} -s 20000101 -r 1 -v  --last-jobtype 
prepcompute --open-run

echo $(date +"%a %b  %e %T %Y") : compute 1 2000-01-01T00:00:00 3521584 - done >
> /scratch/duts/runtime/awicm3-frontiers-xios//testpidbug/log//testpidbug_awicm3
.log

wait

**Comment by nwieters** [comment]
Hi @tsemmler05 ,

can you please give me access rights to the last folder `testpidbug` Thank you.

**Comment by tsemmler05** [comment]
O.k., now you should have access.

**Comment by tsemmler05** [comment]
Mmh, just happened to see that while the .run script looks the same at the end, the model simulation failed at a different place. I guess I have to revert to the state of yesterday (i.e. go back three commits) to reproduce the error of yesterday. Sorry for the confusion.

**Comment by tsemmler05** [comment]
O.k., I have reverted back but now the error doesn't occur anymore. Since I had exactly the same problem some months ago (which resulted in looking for ages why I only got the rstas.nc but not the rstos.nc restart files), I thought I should report the problem. But it seems like with the original setup (which is slow at ECMWF atos) the problem does not occur (only with the optimizations suggested by ECMWF user support to get a reasonable performance with OMP_NUM_THREADS not equal 1). So I guess we have to wait until we fix the run scripts with this optimization (and see if the problem reoccurs). Sorry again for the false alarm.
